### DISCUSSION TITLE
[Question] Are there global hooks for triggering callbacks?

### DISCUSSION THREAD

**Discussion by betweenbrain** [Akryum/vue-apollo#856] [Q&A]
Hello,

I've been researching this, but not found a definitive answer. Are there any global hooks, not component hooks, triggered when an Apollo query is executed, ideally after it completes? That is, I'm looking for something that would be like a callback executed each time a query is completed or when `InMemoryCache` is updated.

Thanks in advance!

**Comment by Akryum** [comment]
Write an Apollo Link

**Comment by betweenbrain** [comment]
> Write an Apollo Link

Thanks! I didn't realize that Apollo Links were like hooks. Seems like different terminology for similar things. 😄

**Comment by Akryum** [comment]
They are more than hooks. For example, the HTTP link is the one allowing you to make HTTP requests to a GraphQL API.

**Comment by betweenbrain** [comment]
Perfect, thanks again! I will start digging into their documentation.
